test_gcs.py

```
@@ -0,0 +1,85 @@
+        
+        try:
+            with open(gcp_keypath, "r") as f:
+                credentials_info = json.load(f)
+            client = storage.Client.from_service_account_info(credentials_info)
+            logger.info(f"GCS client initialized for project: {client.project}")
+        except FileNotFoundError as exc:
+            logger.error(f"GCP key file not found: {gcp_keypath}")
+            raise FileNotFoundError(f"GCP key file not found: {gcp_keypath}") from exc
+        except Exception as e:
+            logger.error(f"Failed to initialize GCS client: {str(e)}")
+            raise RuntimeError(f"Failed to initialize GCS client: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)